cms/tests/test_app_registration.py

```diff
@@ -53,4 +53,186 @@ def test_imports_cms_apps_files(self):
     ])
     def test_raises_exception_raised_in_cms_file(self):
         with self.assertRaises(KeyError):
-            autodiscover_cms_files()
+            app_registration.autodiscover_cms_configs()
+
+    @override_settings(INSTALLED_APPS=[
+        'cms.tests.test_app_registry.app_with_cms_extension',
+        'cms.tests.test_app_registry.app_without_cms_file'
+    ])
+    def test_adds_cms_app_attribute_to_django_app_config(self):
+        app_registration.autodiscover_cms_configs()
+
+        app_list = [app for app in apps.get_app_configs()]
+        self.assertTrue(hasattr(app_list[0], 'cms_app'))
+        self.assertEqual(
+            app_list[0].cms_app.__class__.__name__, 'CMSSomeFeatureConfig')
+        self.assertFalse(hasattr(app_list[1], 'cms_app'))
+
+    @override_settings(INSTALLED_APPS=[
+        'cms.tests.test_app_registry.app_without_cms_app_class',
+    ])
+    def test_raises_exception_when_no_cms_app_class_found_in_cms_file(self):
+        with self.assertRaises(ImproperlyConfigured):
+            app_registration.autodiscover_cms_configs()
+
+    @override_settings(INSTALLED_APPS=[
+        'cms.tests.test_app_registry.app_with_two_cms_app_classes',
+    ])
+    def test_raises_exception_when_more_than_one_cms_app_class_found_in_cms_file(self):
+        with self.assertRaises(ImproperlyConfigured):
+            app_registration.autodiscover_cms_configs()
+
+
+class GetCmsAppsWithFeaturesTestCase(CMSTestCase):
+
+    @patch.object(apps, 'get_app_configs')
+    def test_returns_only_apps_with_features(self, mocked_apps):
+        # apps with cms_app attr and a configure method
+        app_with_cms_ext1 = Mock(cms_app=Mock(spec=['configure_app']))
+        app_with_cms_ext2 = Mock(cms_app=Mock(spec=['configure_app']))
+        # app with cms_app attr without a configure method
+        # this throws an AttributeError if configure_app is accessed
+        app_with_cms_config = Mock(cms_app=Mock(spec=[]))
+        # app without cms_app attr
+        # this throws an AttributeError if cms_app is accessed
+        non_cms_app = Mock(spec=[])
+        # mock what apps have been installed
+        mocked_apps.return_value = [
+            app_with_cms_ext1,
+            app_with_cms_config,
+            app_with_cms_ext2,
+            non_cms_app,
+        ]
+
+        apps_with_features = app_registration.get_cms_apps_with_features()
+
+        self.assertListEqual(
```

Comment:
What is this assertion doing i.e. what is it testing for? A small comment would make it easy for me to understand. Again personal opinion.